srv/salt/ceph/dashboard/default.sls

```diff
@@ -0,0 +1,15 @@
+enable ceph dashboard:
+  cmd.run:
+    - name: ceph mgr module enable dashboard
+    - failhard: True
+
+create self signed certificate:
+  cmd.run:
+    - name: ceph dashboard create-self-signed-cert
+    - failhard: True
+
+set username and password:
+  cmd.run:
+    # This command is printed although the 'unless' statement evaluates as true. This might cause confusion.
+    - name: ceph dashboard ac-user-create {{ salt['pillar.get']('dashboard_username', 'admin') }} {{ salt['pillar.get']('dashboard_password', salt['random.get_str'](10)) }} administrator || true
```

Comment:
We might want to consult with @LenzGr here, since users will want to access the dashboard and this (randomly generated string as password) might make it less-than-trivial to do so?

Reply:
Nobody is asking me, but if they did, I'd advocate for having a hard-coded default password, like "admin", which is documented along with a big fat box advising folks to change that password as soon as possible after the cluster is deployed.

If someone wants security, they can set the password via the pillar, although not sure how secure that is?

Reply:
I'd be ok with a random password but then we'd need to persist is somewhere so it can be looked up. Ideally this would be handled by the dashboard itself: No requirement to add a password and on first login the user is required to add password (but I guess this is a feature request).

Reply:
We had this conversation on rocketchat #ceph (Thursday evening ~6pm).

I should've written the conclusion in the PR body, my bad.

* random generated password are to be favoured over a hard-coded default password.
* the random password needs to be printed to the screen (this is a todo, I'll just add a advise runner)
* the dashboard will get a 'change-password-after-first-login' policy
* you should be allowed to set the password in the pillar (implemented)

Reply:
Hm, then I guess we are going to need documentation on how to reset the dashboard password, since I'm having trouble imagining a 100% success rate on users (a) noticing the random dashboard password printed on the screen and (b) remembering it for long enough that they can use it to log in.

But maybe I'm getting pessimistic in my old age.

Presumably, purging the dashboard role and then redeploying it will reset the password?

Reply:
CC'ing all parties involved @LenzGr @l-mb

Reply:
This just seems like we're setting ourselves up for user disappointment. It may seem trivial to us to (a) notice a randomly generated password on the screen, before it scrolls off the top edge, and (b) be capable of entering that password into the dashboard login page, but the real "user experience" might be quite different (and disappointing).

Reply:
Setting up a default password for an admin account on a product is too risky.

That needs to always be user set or unique to a particular install.